client/src/modules/list/components/ListHeader.jsx

```diff
@@ -0,0 +1,238 @@
+
+      action();
+    }
+  };
+
+  handleClick = event => {
+    const {
+      isDescriptionTextareaVisible,
+      isNameInputVisible,
+      name
+    } = this.state;
+    const { target } = event;
+
+    if (
+      isDescriptionTextareaVisible &&
+      !this.descriptionTextarea.current.contains(target)
+    ) {
+      this.setState({ isDescriptionTextareaVisible: false });
+      this.handleDescriptionUpdate();
+      return;
+    }
+
+    if (
+      isNameInputVisible &&
+      name.trim().length >= 1 &&
+      !this.nameInput.current.contains(target)
+    ) {
+      this.setState({ isNameInputVisible: false });
+      this.handleNameUpdate();
+    }
+  };
+
+  handleNameUpdate = () => {
+    const { name } = this.state;
+    const {
+      details,
+      match: {
+        params: { id }
+      },
+      updateList
+    } = this.props;
+    const { name: previousName } = details;
+
+    if (previousName === name) {
+      this.setState({ isNameInputVisible: false });
+      return;
+    }
+
+    if (name.trim().length >= 1) {
+      updateList(id, { name });
+      this.setState({ isNameInputVisible: false });
+    }
+  };
+
+  handleDescriptionUpdate = () => {
+    const { description } = this.state;
+    const {
+      details,
+      match: {
+        params: { id }
+      },
+      updateList
+    } = this.props;
+    const { description: previousDescription } = details;
+
+    if (previousDescription.trim() === description.trim()) {
+      this.setState({ isDescriptionTextareaVisible: false });
+      return;
+    }
+
+    if (whiteSpaceOnly(description)) {
+      updateList(id, { description: '' });
+      return;
+    }
+
+    updateList(id, { description });
+
+    this.setState({ isDescriptionTextareaVisible: false });
+  };
+
+  render() {
+    const {
+      description,
+      isDescriptionTextareaVisible,
+      isNameInputVisible,
+      name
+    } = this.state;
+
+    return (
+      <div className="list-header">
+        <div className="list-header__top">
+          <ListIcon />
+          {isNameInputVisible ? (
+            <input
+              className="list-header__name-input primary-input"
+              name="name"
+              onChange={this.handleNameChange}
+              ref={this.nameInput}
+              type="text"
+              value={name}
+            />
+          ) : (
+            <h1
+              className="list-header__heading"
+              onClick={this.handleNameInputVisibility}
+            >
+              {name}
+            </h1>
+          )}
+        </div>
+        {isDescriptionTextareaVisible ? (
+          <textarea
+            className="cohort-header__desc-textarea primary-textarea"
+            name="description"
+            onChange={this.handleDescriptionChange}
+            ref={this.descriptionTextarea}
+            type="text"
+            value={description}
+          />
+        ) : (
+          <Fragment>
+            {description && (
```

Comment:
why not use ternary operator here? You've got true and false values checked and the return value is used.

Reply:
Eslint says, do not nest ternary 
https://eslint.org/docs/rules/no-nested-ternary

Should I put ternary here anyway?